api/cds.proto

```diff
@@ -272,4 +272,28 @@ message Cluster {
     repeated LbSubsetSelector subset_selectors = 3;
   }
   LbSubsetConfig lb_subset_config = 22;
+  
+  message RingHashLbConfig {
+    // The hash function used to build the consistent hashing ring by hashing
+    // each server's address.
+    enum HashFunction {
+      // Standard library
+      STL = 0;
+      // XXHash
+      XXH64 = 1;
+    }
+    HashFunction hash_function = 1;
+
+    // Minimum hash ring size, i.e. total virtual nodes. A larger size
+    // will provide better request distribution since each host in the
+    // cluster will have more virtual nodes.
+    google.protobuf.UInt64Value minimum_ring_size = 2;
+  }
+
+  // Optional configuration for the load balancing algorithm selected by
+  // LbPolicy. Currently only RING_HASH has additional configuration
+  // options.
```

Comment:
nit: comment that it is an error to specify a RingHash config when not using ring hash LB.